File: 53-maximum-subarray/53-maximum-subarray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int curr = nums[0], maxSub = curr;
        for (int i = 1; i < nums.size(); ++i) {
            curr = max(nums[i], curr + nums[i]);
            maxSub = max(curr, maxSub);
        }
        return maxSub;
    }
};

File: 53-maximum-subarray/53-maximum-subarray.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        currSub, maxSub = nums[0], nums[0];
        for number in nums[1:]:
            currSub = max(number, currSub + number)
            maxSub = max(currSub, maxSub)
        return maxSub